src/core/init/load_on_media_source.ts

```diff
@@ -184,6 +186,15 @@ export default function createMediaSourceLoader({
               handleDiscontinuity(gap[1], mediaElement);
             }
             return EMPTY;
+          case "addedSegmentOnAdaptationChange":
```

Comment:
Maybe we could return an event that explicitly asks for a flush ? As needsMediaSourceReload, which asks for the reloading.

Reply:
Ho yea got the thing.

I dont fire an `addedSegmentOnAdaptationChange` anymore but rather a `needsSourceBufferFlush`.
 As mentioned below not sure about the name though?

Also, I'm firing that event even when we are on `audioTrackSwitchingMode: smooth|flush` but not on `audioTrackSwitchingMode: reload`
Is it something that I should take of? Since we are not doing any action on `smooth` strategy?